docs/source/reference/python27-eol.rst

```diff
@@ -2,19 +2,42 @@
+As per `PEP 373 <http://legacy.python.org/dev/peps/pep-0373/>`_, Python 2.7
+`will cease to be supported in 2020 <https://pythonclock.org/>`_.  Like many
+projects, the Numba team has to consider how to time its own end of Python 2.7
+support.  Given how deeply Numba must interface with the Python interpreter,
+supporting both Python 2 and 3 creates quite a development and testing burden.
+In addition, Numba (specifically via llvmlite) has to deal with some tricky
+compiler issues on Windows, where LLVM requires Visual Studio 2015 or later,
+but Python 2.7 extensions must be built with Visual Studio 2008.  Needless to
+say, the goal with this plan is to support our Python 2.7 user base (~30% of
+downloads as of February 2018), but also clearly signal that *now is the time
+to switch to Python 3 if you have not already*.
 
-Python 2.7 users of Numba should also be aware of `NumPy's timeline for ending Python 2.7 support <https://github.com/numpy/numpy/blob/master/doc/neps/nep-0014-dropping-python2.7-proposal.rst>`_.  Due to Numba's tight coupling with NumPy, the NumPy timeline has strongly informed the Numba timeline below.
+Python 2.7 users of Numba should also be aware of `NumPy's timeline for ending Python 2.7 support
+<https://github.com/numpy/numpy/blob/master/doc/neps/nep-0014-dropping-python2.7-proposal.rst>`_.
+Due to Numba's tight coupling with NumPy, the NumPy timeline has 
+informed the Numba timeline below.
 
 Timeline
 ========
 
+This timeline was revised in August 2019 to reflect the reality that we wanted
+to get more improvements into Numba before ending Python 2.7 support.  Note
+that the Numba versions mentioned below are more concrete than the dates.
+
 The end of Python 2.7 support in Numba will be staged:
 
-* **December 2018**: Tag and release Numba 1.x.0.  Create a Python 2.7 branch based on this release.  
-* Critical fixes until **January 1, 2020** will be backported to the Python 2.7 branch and released as Numba 1.x.y.
-* No new features will be added to the Python 2.7 branch, but we will continue to do automated testing of it with new NumPy releases.
-* **January 1, 2019**: we will slim down the Numba master branch by removing all the Python 2.7 compatibility code and release Numba 1.(x+1).0, which will be functionally identical to Numba 1.x.0.
-* **January 1, 2020**: The Numba developers will stop supporting the Python 2.7 branch.
+* **Numba 0.47**: (~Dec 2019) This is will be the last version of Numba to
+  support both Python 2 and 3.
+* **Numba 0.48 development cycle**: (Jan-Mar 2020) Python 2 support will be 
+  removed from ``master`` branch. If any critical bug fixes are discovered in
+  the 0.47.0 release, they will be backported to create 0.47.x patch releases.
+* **Numba 0.48**: (~Mar 2020) This will be the first release of Numba that is
+  Python 3 only.  We will set the metadata on the package to require Python
+  3.5 or later.
```

Comment:
Have you considered or discussed the variant with a minimum version of 3.6 instead of 3.5? I ask for two reasons, firstly, I got the impression that many projects (numpy, pandas, matplotlib, ...) are planning to stop supporting 3.5 (along with Python 2) around the beginning of the 2020. Secondly, because 3.6 has several rather useful features, especially for library developers, like f-strings, dict-ordering, ... Of course, I’m more likely as an outside viewer, but it seems to me that this can be an unreasonable waste of yours time, not to immediately use the 3.6 capabilities during removal of Python 2 compatibility code,  only for 1-2 releases with 3.5 support.

Reply:
We are tracking the discussion around NEP 29, which would make Python 3.6 the minimum by the time we get to our first 2020 release.  I didn't want to commit to that here while it is still being discussed.  I'm a little nervous about dropping Python 3.5 (esp. for slow to update systems like the Raspberry Pi), and none of the Python 3.6 features are so compelling for us that we want to rush it.